src/Core/src/Platform/iOS/ImageAnimationHelper.cs

```diff
@@ -0,0 +1,254 @@
+		}
+
+		for (int i = 0; i < imageCount; i++)
+		{
+			imageData.AddFrameData(i, imageSource);
+		}
+
+		MauiCAKeyFrameAnimation animation = imageData.CreateKeyFrameAnimation(imageSource);
+
+		if (animation != null)
+		{
+			animation.RemovedOnCompletion = false;
+			animation.KeyPath = "contents";
+			animation.RepeatCount = repeatCount;
+			animation.Width = imageData.Width;
+			animation.Height = imageData.Height;
+
+			if (imageCount == 1)
+			{
+				animation.Duration = double.MaxValue;
+				animation.KeyTimes = null;
+			}
+		}
+
+		return animation;
+	}
+
+	public static Task<MauiCAKeyFrameAnimation> CreateAnimationFromImageSource(IImageSource imageSource, CancellationToken cancellationToken = default)
+	{
+		if (imageSource is IStreamImageSource streamImageSource)
+		{
+			return CreateAnimationFromStreamImageSourceAsync(streamImageSource, cancellationToken);
+		}
+		else if (imageSource is IFileImageSource fileImageSource)
+		{
+			return Task.FromResult(CreateAnimationFromFileImageSource(fileImageSource));
+		}
+		else if (imageSource is IUriImageSource uriImageSource)
+		{
+			return CreateAnimationFromUriImageSourceAsync(uriImageSource, cancellationToken);
+		}
+
+		throw new ArgumentException($"Type of {nameof(imageSource)} is not supported for loading animations.");
+	}
+
+	public static async Task<MauiCAKeyFrameAnimation> CreateAnimationFromStreamImageSourceAsync(IStreamImageSource imageSource, CancellationToken cancellationToken = default)
+	{
+		MauiCAKeyFrameAnimation animation = null;
+
+        using var streamImage = await imageSource.GetStreamAsync(cancellationToken).ConfigureAwait(false);
+		if (streamImage != null)
+		{
+			using var parsedImageSource = CGImageSource.FromData(NSData.FromStream(streamImage));
+			animation = CreateAnimationFromCGImageSource(parsedImageSource);
+		}
+
+		return animation;
+	}
+
+	public static async Task<MauiCAKeyFrameAnimation> CreateAnimationFromUriImageSourceAsync(IUriImageSource imageSource, CancellationToken cancellationToken = default)
+	{
+		MauiCAKeyFrameAnimation animation = null;
+
+		if (imageSource?.Uri != null)
+		{
+			using var streamImage = await ((IStreamImageSource)imageSource).GetStreamAsync(cancellationToken).ConfigureAwait(false);
+			if (streamImage != null)
+			{
+				using (var parsedImageSource = CGImageSource.FromData(NSData.FromStream(streamImage)))
+				{
+					animation = CreateAnimationFromCGImageSource(parsedImageSource);
+				}
+			}
+		}
+
+		return animation;
+	}
+
+	public static MauiCAKeyFrameAnimation CreateAnimationFromFileImageSource(IFileImageSource imageSource)
+	{
+		MauiCAKeyFrameAnimation animation = null;
+		string file = imageSource?.File;
+
+		if (!string.IsNullOrEmpty(file))
+		{
+
+			var root = NSBundle.MainBundle.BundlePath;
+#if MACCATALYST
+			root = Path.Combine(root, "Contents", "Resources");
+#endif
+			using var parsedImageSource = CGImageSource.FromUrl(NSUrl.CreateFileUrl(Path.Combine(root, file), null));
```

Comment:
Ah looking here you are already calling the services and using that. What if this logic was in the service so it just flowed through? Did you have issues with that?